torchgen/_autoheuristic/train_decision.py

```diff
@@ -102,8 +103,19 @@ def predict(self, model, df, feature_columns):
         leaf_ids = model.apply(df[feature_columns])
         return predictions, proba, leaf_ids
 
+    def ranking_num_choices(self):
+        # if the heuristic is used for ranking, this function returns the number
+        # of choices that the heuristic will return
+        return 5
```

Comment:
I'm surprised this is hardcoded. Is this not configurable at all ?

Reply:
You can configure it by overriding the method https://github.com/pytorch/pytorch/pull/131714/files#diff-45bd9b9ea64020450836e3496f0da071f133e1065dc62fbbb141bd6f52d6d2b0R69. But I'll make this a command line option instead.